src/main/java/net/neoforged/neoforge/registries/DeferredBlock.java

```diff
@@ -18,6 +19,24 @@
  * @param <T> The specific {@link Block} type.
  */
 public class DeferredBlock<T extends Block> extends DeferredHolder<Block, T> implements ItemLike {
+    /**
+     * Creates a new {@link ItemStack} with a default size of 1 from this {@link Item}
+     */
+    public ItemStack asItemStack() {
+        return asItemStack(1);
+    }
+
+    /**
+     * Creates a new {@link ItemStack} with the given size from this {@link Item}
+     *
+     * @param count The size of the stack to create
+     */
+    public ItemStack asItemStack(int count) {
+        ItemStack stack = asItem().getDefaultInstance();
```

Comment:
This should probably throw if the stack would be empty.

Reply:
fixed!